ami/trainers/i_jepa_latent_visualization_trainer.py

```diff
@@ -0,0 +1,217 @@
+
+    def _is_new_data_available(self) -> bool:
+        return (
+            self.image_data_user.buffer.count_data_added_since(self.dataset_previous_get_time)
+            >= self.minimum_new_data_count
+        )
+
+    def get_dataset(self) -> Dataset[Tensor]:
+        dataset = self.image_data_user.get_dataset()
+        self.dataset_previous_get_time = time.time()
+        return dataset
+
+    @torch.no_grad()
+    def log_visualization(self, dataloader: DataLoader[Tensor]) -> None:
+        """logs grid of input and reconstructed images for visualization.
+
+        Args:
+            dataloader (DataLoader[Tensor]): DataLoader containing the image batches.
+        """
+        image_batches = []
+        reconstruction_image_batches = []
+        num_remaining = self.num_visualize_images
+
+        batch: tuple[Tensor]
+        for batch in dataloader:
+            (image_batch,) = batch
+            image_batch = image_batch[:num_remaining]
+            image_batches.append(image_batch)
+            image_batch = image_batch.to(self.device)
+
+            latents = self.encoder(image_batch)
+            reconstruction: Tensor = self.decoder(latents)
+
+            reconstruction_image_batches.append(reconstruction.cpu())
+
+            num_remaining -= reconstruction.size(0)
+            if num_remaining <= 0:
+                break
+
+        grid_input_image = torchvision.utils.make_grid(torch.cat(image_batches), self.visualize_grid_row)
+        grid_reconstruction_image = torchvision.utils.make_grid(
+            torch.cat(reconstruction_image_batches), self.visualize_grid_row
+        )
+
+        self.logger.tensorboard.add_image(self.log_prefix + "metrics/input", grid_input_image, self.logger.global_step)
+        self.logger.tensorboard.add_image(
+            self.log_prefix + "metrics/reconstruction", grid_reconstruction_image, self.logger.global_step
+        )
+
+    @override
+    def train(self) -> None:
+        # move to device
+        self.encoder.to(self.device)
+        self.decoder.to(self.device)
+
+        optimizer = self.partial_optimizer(self.decoder.parameters())
+        optimizer.load_state_dict(self.optimizer_state)
+
+        # prepare about dataset
+        dataloader = self.partial_dataloader(dataset=self.get_dataset())
+
+        for _ in range(self.max_epochs):
+            batch: tuple[
+                Tensor,
+            ]
+            for batch in dataloader:
+                (image_batch,) = batch
+                image_batch = image_batch.to(self.device)
+
+                with torch.no_grad():
+                    latents = self.encoder(image_batch)
+                    # latents: [batch_size, n_patches_height * n_patches_width, latents_dim]
+
+                image_out: Tensor = self.decoder(latents)
+                image_size = image_out.size()[-2:]
+
+                image_batch_resized = torchvisF.resize(image_batch, image_size)
```

Comment:
`torchvisF`のimport文の変更に合わせてこっちも変更
```suggestion
                image_batch_resized = torchvision.transforms.v2.functional.resize(image_batch, image_size)
```

Reply:
OK